### PR TITLE
Provide --sclize option

### PR DESCRIPTION
**Pull request by irushchyshyn**
Enable generating sclized spec files using spec2scl.
The following spec2scl options can be provided alongside:

- --no-meta-runtime-dep
- --no-meta-buildtime-dep
- --skip-functions
- --no-deps-convert
- --list-file

Issue #100

### CODE REVIEW

File: pyp2rpm/bin.py [hunk 1]
```diff
@@ -8,6 +8,10 @@
 from pyp2rpm.logger import register_file_log_handler, register_console_log_handler
 
 import click
+try:
+    from spec2scl.convertor import Convertor as SclConvertor
+except:
```
Comment (mcyprian) [inline pyp2rpm/bin.py]: Can something except ImportError (ModuleNotFoundError) occur here? Maybe we could log the information when something unexpected happens.

File: pyp2rpm/bin.py [hunk 2]
```diff
@@ -98,6 +126,11 @@ def main(package, v, d, s, r, proxy, srpm, p, b, o, t, venv):
     converted = convertor.convert()
     logger.debug('Convertor: {0} succesfully converted.'.format(convertor))
 
+    if sclize:
+        converted = convert_to_scl(converted, scl_kwargs)
+        if not converted:
+            return
```
Comment (mcyprian) [inline pyp2rpm/bin.py]: I caught the idea of this part, but at the beginning it was not clear to me in which cases is the return value of convert_to_scl None/Empty string. Can you add a short comment please.

**Review by mcyprian — CHANGES_REQUESTED** [review]
Simple and elegant solution, a few things could be improved in my opinion:
- when running pyp2rpm package --no-meta-runtime-dep it seems this option is simply ignored. Combining these options with --sclize should be required or user should be warn that option has no effect in this case.
- I know that help is autogenerated in click, but isn't it possible to make the fact that the last 4 options are SCL specific a bit more obvious in the help message?
- pyp2rpm's man page (file pyp2rpm.1 in the main project directory) should be updated as well. It should be possible to generate man page from the help message directly, but I've never tried how it works https://www.gnu.org/software/help2man/

**Comment by irushchyshyn** [comment]
Thank you for the review @mcyprian ! 

>     when running pyp2rpm package --no-meta-runtime-dep it seems this option is simply ignored. Combining these options with --sclize should be required or user should be warn that option has no effect in this case.
I have added a validation for a case like that.

>     I know that help is autogenerated in click, but isn't it possible to make the fact that the last 4 options are SCL specific a bit more obvious in the help message?
I have modified help formatting for options, to put SCL related options in a separate options group. Please let me know what you think and how it looks now.

    pyp2rpm's man page (file pyp2rpm.1 in the main project directory) should be updated as well. It should be possible to generate man page from the help message directly, but I've never tried how it works https://www.gnu.org/software/help2man/
Updated.

File: pyp2rpm/bin.py (hunk 1 above)
Comment (irushchyshyn) [reply pyp2rpm/bin.py]: Added an extra check in f83a092

File: pyp2rpm/bin.py (hunk 2 above)
Comment (irushchyshyn) [reply pyp2rpm/bin.py]: I reworked the validation a bit, so this is not needed anymore.

**Comment by mcyprian** [comment]
Could you please also update SYNOPSIS and AUTHOR sections of the man page? The rest of the updates look good to me :)

**Review by mcyprian — APPROVED** [review]

**Comment by irushchyshyn** [comment]
Updated in eaeed79.
Sorry for that, did not notice that it was also changed.

**Pull request merged**